moc_openstack_api_reporting.py

```diff
@@ -344,34 +359,50 @@ def convert_to_ceilometer_event_types(event):
         # "compute.instance.lock": [""],
         # "compute.instance.detach_interface": [""],
         # "compute.instance.rebuild": [""],
-        # "compute.instance.resize": [""],
+        "compute.instance.resize": ["start", "end"],
         # These events are from the reference file
         "compute.instance.create": ["start", "end"],
-        "compute.instance.delete": ["start", "end"],  # does this one have a "end"?
-        # "compute.instance.live_migration._post": ["start", "end"],
-        # "compute.instance.live_migration.post.dest": ["start", "end"],
-        # "compute.instance.live_migration.pre": ["start", "end"],
+        "compute.instance.delete": ["start", "end"],
+        # live_migration is probably not handled by xdmod
+        "compute.instance.live_migration": [
+            "pre.start",
+            "pre.end",
+            "_post.start",
+            "_post.end",
+            "post.dest.start",
+            "post.dest.edn",
+        ],
         "compute.instance.power_off": ["start", "end"],
         "compute.instance.power_on": ["start", "end"],
         "compute.instance.shutdown": ["start", "end"],
     }
+    if event_type not in ceilometer_event_types:
+        return [event_type]
+    event_list = []
+    for event_postfix in ceilometer_event_types[event_type]:
+        event_list.append(event_type + "." + event_postfix)
+    return event_list
+
+
+def convert_to_ceilometer_event_types(event):
+    """maps from a currently defined event to event(s) that were previously defined"""
     if event["event_type"] == "compute.instance.create" and event["state"] == "ERROR":
         event["event_type"] = "compute.instance.create.error"
         return [event]
-    if event["event_type"] in map_current_to_old:
-        event["event_type"] = map_current_to_old[event["event_type"]]
-    if event["event_type"] in ceilometer_event_types:
-        ret_list = []
-        delta_time = 0
-        time_0 = datetime.datetime.fromisoformat(event["generated"])
-        for ceilometer_event in ceilometer_event_types[event["event_type"]]:
-            new_event = copy.deepcopy(event)
-            new_event["event_type"] = f"{new_event['event_type']}.{ceilometer_event}"
-            new_event["generated"] = (time_0 + datetime.timedelta(seconds=delta_time)).isoformat()
-            ret_list.append(new_event)
-            delta_time += 10
-        return ret_list
-    return []
+
+    old_event_name = convert_new_to_old_eventtype(event["event_type"])
+    event_list = get_list_of_ceilometer_event_types(old_event_name)
+
+    ret_list = []
+    delta_time = 0
+    time_0 = datetime.datetime.fromisoformat(event["generated"])
+    for ceilometer_event in event_list:
+        new_event = copy.deepcopy(event)
+        new_event["event_type"] = ceilometer_event
+        new_event["generated"] = (time_0 + datetime.timedelta(seconds=delta_time)).isoformat()
+        ret_list.append(new_event)
+        delta_time += 10
```

Comment:
Yes.  Since many of the single current events get split in to event.start and event.stop and it makes sense that the most rational way of ordering them is by time.  Furthermore the dataset that I analyzed had about a 10 second delta between start and stop of the creation events, I just went with the 10 second delta.

I am not 100% sure if xdmod actually uses this, however, it would not shock me if they did and that was one of the reasons why the events were not showing up in the interface, despite being processed.